src/main/java/io/jenkins/plugins/junit/storage/database/DatabaseTestResultStorage.java

```diff
@@ -874,9 +938,19 @@ public TestResult getResultByNodes(@NonNull List<String> nodeIds) {
     }
 
     private static Span createSpan(String spanName) {
-        Span span = getTracer()
-                .spanBuilder(spanName)
-                .startSpan();
+        return createSpan(spanName, null);
+    }
+
+    private static Span createSpan(String spanName, OtelInformation otelInformation) {
+        Tracer tracer = getTracer();
+        SpanBuilder spanBuilder = tracer.spanBuilder(spanName);
+
+        if (otelInformation != null) {
+            spanBuilder = spanBuilder.setParent(otelInformation.setup())
```

Comment:
@timja I suspect there is a problem in the way the Jenkins OTel Plugin propagates the context if your span lacks a parent. Can we verify the solution doesn't reside in the Jenkins OTel Plugin before rolling out this solution?

Reply:
Yes no rush to roll it out, can wait

Reply:
Problem understood, context propagation is not done by the Jenkins otel plugin to step execution of other plugins.
I'll try to fix it in the Jenkins otel plugin